rat/__init__.py

```diff
@@ -59,16 +196,156 @@
+
+| `skill[team, year]` | `skill[team, shift(year, 1)]` |
+| --------------------------------------------------- |
+| `skill[CHA, 2016]`  | `0`                           |
+| `skill[CHA, 2017]`  | `skill[CHA, 2016]`            |
+| `skill[ATL, 2017]`  | `0`                           |
+
+A negative shift produces a different result:
+
+| `skill[team, year]` | `skill[team, shift(year, -1)]` |
+| --------------------------------------------------- |
+| `skill[CHA, 2016]`  | `skill[CHA, 2017]`            |
+| `skill[CHA, 2017]`  | `0`                           |
+| `skill[ATL, 2017]`  | `0`                           |
+
+## Subscripts
+
+In general, subscripts are resolved and generated
+when found on the **right-hand-side of a statement that has a left-hand-side
+with resolved subscripts**. However since rat topologically sorts before
+evaluation accordingly, the user just has to make sure parameters that they
+have declared/assigned with subscripts must be resolvable by checking they
+are present as right-hand-side.
+
+
+## Distributions
+
+$\mathcal{R}$ here means all real numbers and $\mathcal{R}^+$ means real numbers greater than zero.
+
+
+| Distribution | Constraints |
+| ---------------------------|
+| `y ~ bernoulli_logit(logit_p)` | $y = 0$ or $y = 1$, $\text{logit_p} \in \mathcal{R}$ |
+| `y ~ cauchy(location, scale)` | $y, \text{location} \in \mathcal{R}$, $\text{scale} \in \mathcal{R}^+$ |
+| `y ~ exponential(scale)` | $y, \text{scale} \in \mathcal{R}^+$ |
+| `y ~ log_normal(mu, sigma)` | $\text{mu} \in \mathcal{R}$, $y, \text{sigma} \in \mathcal{R}^+$ |
+| `y ~ normal(mu, sigma)` | $y, \text{mu} \in \mathcal{R}$, $\text{sigma} \in \mathcal{R}^+$ |
+
+## Functions
+
+* `abs(x)`
+* `arccos(x)`
+* `arcsin(x)`
+* `arctan(x)`
+* `ceil(x)`
+* `cos(x)`
+* `exp(x)`
+* `floor(x)`
+* `inverse_logit(x)`
+* `log(x)`
+* `logit(x)`
+* `round(x)`
+* `sin(x)`
+* `tan(x)`
+
+## Operator Precedence Table
+
+|  Operator   |   Precedence   |
+|------|:-----:|
+| function calls(`exp`, `log`, etc.) | 100, leftmost derivative  |
+| prefix negation(`-10`, `-(1+2)`, etc.) | 50 |
+| `^`  | 40  |
+| `*`, `/`, `%`  | 30  |
+| `+`, `-`  | 10  |
+
+# Installation and Use
+
+Rat is only available from Github:
+
 ```
+git clone https://github.com/bbbales2/regressions
+cd regressions
+pip install .
+```
+
+## CLI
 
-The only new line of code is line 2, which uses assignment to declare `theta` with subscript `school`. However the subscript resolving scenario is exactly the same as the previous example: `school` is resolved at line 1 from the dataframe. `z`, the unit normal parameter's subscript `school` is resolved from `theta[school]`.
+Rat is a Python library, but comes with a helper script `rat` to quickly compile and
+run models.
+
+For example, to fit a model `mrp.rat` with the data `mrp.csv` and save the results in
+`output` we can do:
+
+```
+rat mrp.rat mrp.csv output
+```
+
+Type `rat -h` for full usage info.
+"""
+
+__all__ = ["model", "fit"]
```

Comment:
I only import these two modules by default so these are the only two pdoc renders docs for